File: retaguarda-software-house/java/retaguarda_sh/src/main/java/com/t2ti/retaguarda_sh/model/nfe/NfeConfiguracao.java
/*******************************************************************************
Title: T2Ti ERP 3.0                                                                
Description: Model relacionado à tabela [NFE_CONFIGURACAO] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2021 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
package com.t2ti.retaguarda_sh.model.nfe;

import java.io.Serializable;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.NamedQuery;
import javax.persistence.Table;
import java.math.BigDecimal;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.t2ti.retaguarda_sh.model.cadastros.Empresa;

@Entity
@Table(name="NFE_CONFIGURACAO")
@NamedQuery(name="NfeConfiguracao.findAll", query="SELECT t FROM NfeConfiguracao t")
public class NfeConfiguracao implements Serializable {
	private static final long serialVersionUID = 1L;

    @Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Integer id;

    @Column(name="CERTIFICADO_DIGITAL_SERIE")
	private String certificadoDigitalSerie;

    @Column(name="CERTIFICADO_DIGITAL_CAMINHO")
	private String certificadoDigitalCaminho;

    @Column(name="CERTIFICADO_DIGITAL_SENHA")
	private String certificadoDigitalSenha;

    @Column(name="TIPO_EMISSAO")
	private Integer tipoEmissao;

    @Column(name="FORMATO_IMPRESSAO_DANFE")
	private Integer formatoImpressaoDanfe;

    @Column(name="PROCESSO_EMISSAO")
	private Integer processoEmissao;

    @Column(name="VERSAO_PROCESSO_EMISSAO")
	private String versaoProcessoEmissao;

    @Column(name="CAMINHO_LOGOMARCA")
	private String caminhoLogomarca;

    @Column(name="SALVAR_XML")
	private String salvarXml;

    @Column(name="CAMINHO_SALVAR_XML")
	private String caminhoSalvarXml;

    @Column(name="CAMINHO_SCHEMAS")
	private String caminhoSchemas;

    @Column(name="CAMINHO_ARQUIVO_DANFE")
	private String caminhoArquivoDanfe;

    @Column(name="CAMINHO_SALVAR_PDF")
	private String caminhoSalvarPdf;

    @Column(name="WEBSERVICE_UF")
	private String webserviceUf;

    @Column(name="WEBSERVICE_AMBIENTE")
	private Integer webserviceAmbiente;

    @Column(name="WEBSERVICE_PROXY_HOST")
	private String webserviceProxyHost;

    @Column(name="WEBSERVICE_PROXY_PORTA")
	private Integer webserviceProxyPorta;

    @Column(name="WEBSERVICE_PROXY_USUARIO")
	private String webserviceProxyUsuario;

    @Column(name="WEBSERVICE_PROXY_SENHA")
	private String webserviceProxySenha;

    @Column(name="WEBSERVICE_VISUALIZAR")
	private String webserviceVisualizar;

    @Column(name="EMAIL_SERVIDOR_SMTP")
	private String emailServidorSmtp;

    @Column(name="EMAIL_PORTA")
	private Integer emailPorta;

    @Column(name="EMAIL_USUARIO")
	private String emailUsuario;

    @Column(name="EMAIL_SENHA")
	private String emailSenha;

    @Column(name="EMAIL_ASSUNTO")
	private String emailAssunto;

    @Column(name="EMAIL_AUTENTICA_SSL")
	private String emailAutenticaSsl;

    @Column(name="EMAIL_TEXTO")
	private String emailTexto;

    @Column(name="NFCE_ID_CSC")
	private String nfceIdCsc;

    @Column(name="NFCE_CSC")
	private String nfceCsc;

    @Column(name="NFCE_MODELO_IMPRESSAO")
	private String nfceModeloImpressao;

    @Column(name="NFCE_IMPRIMIR_ITENS_UMA_LINHA")
	private String nfceImprimirItensUmaLinha;

    @Column(name="NFCE_IMPRIMIR_DESCONTO_POR_ITEM")
	private String nfceImprimirDescontoPorItem;

    @Column(name="NFCE_IMPRIMIR_QRCODE_LATERAL")
	private String nfceImprimirQrcodeLateral;

    @Column(name="NFCE_IMPRIMIR_GTIN")
	private String nfceImprimirGtin;

    @Column(name="NFCE_IMPRIMIR_NOME_FANTASIA")
	private String nfceImprimirNomeFantasia;

    @Column(name="NFCE_IMPRESSAO_TRIBUTOS")
	private String nfceImpressaoTributos;

    @Column(name="NFCE_MARGEM_SUPERIOR")
	private BigDecimal nfceMargemSuperior;

    @Column(name="NFCE_MARGEM_INFERIOR")
	private BigDecimal nfceMargemInferior;

    @Column(name="NFCE_MARGEM_DIREITA")
	private BigDecimal nfceMargemDireita;

    @Column(name="NFCE_MARGEM_ESQUERDA")
	private BigDecimal nfceMargemEsquerda;

    @Column(name="NFCE_RESOLUCAO_IMPRESSAO")
	private Integer nfceResolucaoImpressao;

    @Column(name="NFCE_TAMANHO_FONTE_ITEM")
	private Integer nfceTamanhoFonteItem;

    @Column(name="RESP_TEC_CNPJ")
	private String respTecCnpj;

    @Column(name="RESP_TEC_CONTATO")
	private String respTecContato;

    @Column(name="RESP_TEC_EMAIL")
	private String respTecEmail;

    @Column(name="RESP_TEC_FONE")
	private String respTecFone;

    @Column(name="RESP_TEC_ID_CSRT")
	private String respTecIdCsrt;

    @Column(name="RESP_TEC_HASH_CSRT")
	private String respTecHashCsrt;

    @ManyToOne
	@JsonIgnore
	@JoinColumn(name="ID_EMPRESA")
    private Empresa empresa;

	public NfeConfiguracao() {
		//construtor padrão
	}

    public Integer getId() {
		return this.id;
	}

	public void setId(Integer id) {
		this.id = id;
	}

    public String getCertificadoDigitalSerie() {
		return this.certificadoDigitalSerie;
	}

	public void setCertificadoDigitalSerie(String certificadoDigitalSerie) {
		this.certificadoDigitalSerie = certificadoDigitalSerie;
	}

    public String getCertificadoDigitalCaminho() {
		return this.certificadoDigitalCaminho;
	}

	public void setCertificadoDigitalCaminho(String certificadoDigitalCaminho) {
		this.certificadoDigitalCaminho = certificadoDigitalCaminho;
	}

    public String getCertificadoDigitalSenha() {
		return this.certificadoDigitalSenha;
	}

	public void setCertificadoDigitalSenha(String certificadoDigitalSenha) {
		this.certificadoDigitalSenha = certificadoDigitalSenha;
	}

    public Integer getTipoEmissao() {
		return this.tipoEmissao;
	}

	public void setTipoEmissao(Integer tipoEmissao) {
		this.tipoEmissao = tipoEmissao;
	}

    public Integer getFormatoImpressaoDanfe() {
		return this.formatoImpressaoDanfe;
	}

	public void setFormatoImpressaoDanfe(Integer formatoImpressaoDanfe) {
		this.formatoImpressaoDanfe = formatoImpressaoDanfe;
	}

    public Integer getProcessoEmissao() {
		return this.processoEmissao;
	}

	public void setProcessoEmissao(Integer processoEmissao) {
		this.processoEmissao = processoEmissao;
	}

    public String getVersaoProcessoEmissao() {
		return this.versaoProcessoEmissao;
	}

	public void setVersaoProcessoEmissao(String versaoProcessoEmissao) {
		this.versaoProcessoEmissao = versaoProcessoEmissao;
	}

    public String getCaminhoLogomarca() {
		return this.caminhoLogomarca;
	}

	public void setCaminhoLogomarca(String caminhoLogomarca) {
		this.caminhoLogomarca = caminhoLogomarca;
	}

    public String getSalvarXml() {
		return this.salvarXml;
	}

	public void setSalvarXml(String salvarXml) {
		this.salvarXml = salvarXml;
	}

    public String getCaminhoSalvarXml() {
		return this.caminhoSalvarXml;
	}

	public void setCaminhoSalvarXml(String caminhoSalvarXml) {
		this.caminhoSalvarXml = caminhoSalvarXml;
	}

    public String getCaminhoSchemas() {
		return this.caminhoSchemas;
	}

	public void setCaminhoSchemas(String caminhoSchemas) {
		this.caminhoSchemas = caminhoSchemas;
	}

    public String getCaminhoArquivoDanfe() {
		return this.caminhoArquivoDanfe;
	}

	public void setCaminhoArquivoDanfe(String caminhoArquivoDanfe) {
		this.caminhoArquivoDanfe = caminhoArquivoDanfe;
	}

    public String getCaminhoSalvarPdf() {
		return this.caminhoSalvarPdf;
	}

	public void setCaminhoSalvarPdf(String caminhoSalvarPdf) {
		this.caminhoSalvarPdf = caminhoSalvarPdf;
	}

    public String getWebserviceUf() {
		return this.webserviceUf;
	}

	public void setWebserviceUf(String webserviceUf) {
		this.webserviceUf = webserviceUf;
	}

    public Integer getWebserviceAmbiente() {
		return this.webserviceAmbiente;
	}

	public void setWebserviceAmbiente(Integer webserviceAmbiente) {
		this.webserviceAmbiente = webserviceAmbiente;
	}

    public String getWebserviceProxyHost() {
		return this.webserviceProxyHost;
	}

	public void setWebserviceProxyHost(String webserviceProxyHost) {
		this.webserviceProxyHost = webserviceProxyHost;
	}

    public Integer getWebserviceProxyPorta() {
		return this.webserviceProxyPorta;
	}

	public void setWebserviceProxyPorta(Integer webserviceProxyPorta) {
		this.webserviceProxyPorta = webserviceProxyPorta;
	}

    public String getWebserviceProxyUsuario() {
		return this.webserviceProxyUsuario;
	}

	public void setWebserviceProxyUsuario(String webserviceProxyUsuario) {
		this.webserviceProxyUsuario = webserviceProxyUsuario;
	}

    public String getWebserviceProxySenha() {
		return this.webserviceProxySenha;
	}

	public void setWebserviceProxySenha(String webserviceProxySenha) {
		this.webserviceProxySenha = webserviceProxySenha;
	}

    public String getWebserviceVisualizar() {
		return this.webserviceVisualizar;
	}

	public void setWebserviceVisualizar(String webserviceVisualizar) {
		this.webserviceVisualizar = webserviceVisualizar;
	}

    public String getEmailServidorSmtp() {
		return this.emailServidorSmtp;
	}

	public void setEmailServidorSmtp(String emailServidorSmtp) {
		this.emailServidorSmtp = emailServidorSmtp;
	}

    public Integer getEmailPorta() {
		return this.emailPorta;
	}

	public void setEmailPorta(Integer emailPorta) {
		this.emailPorta = emailPorta;
	}

    public String getEmailUsuario() {
		return this.emailUsuario;
	}

	public void setEmailUsuario(String emailUsuario) {
		this.emailUsuario = emailUsuario;
	}

    public String getEmailSenha() {
		return this.emailSenha;
	}

	public void setEmailSenha(String emailSenha) {
		this.emailSenha = emailSenha;
	}

    public String getEmailAssunto() {
		return this.emailAssunto;
	}

	public void setEmailAssunto(String emailAssunto) {
		this.emailAssunto = emailAssunto;
	}

    public String getEmailAutenticaSsl() {
		return this.emailAutenticaSsl;
	}

	public void setEmailAutenticaSsl(String emailAutenticaSsl) {
		this.emailAutenticaSsl = emailAutenticaSsl;
	}

    public String getEmailTexto() {
		return this.emailTexto;
	}

	public void setEmailTexto(String emailTexto) {
		this.emailTexto = emailTexto;
	}

    public String getNfceIdCsc() {
		return this.nfceIdCsc;
	}

	public void setNfceIdCsc(String nfceIdCsc) {
		this.nfceIdCsc = nfceIdCsc;
	}

    public String getNfceCsc() {
		return this.nfceCsc;
	}

	public void setNfceCsc(String nfceCsc) {
		this.nfceCsc = nfceCsc;
	}

    public String getNfceModeloImpressao() {
		return this.nfceModeloImpressao;
	}

	public void setNfceModeloImpressao(String nfceModeloImpressao) {
		this.nfceModeloImpressao = nfceModeloImpressao;
	}

    public String getNfceImprimirItensUmaLinha() {
		return this.nfceImprimirItensUmaLinha;
	}

	public void setNfceImprimirItensUmaLinha(String nfceImprimirItensUmaLinha) {
		this.nfceImprimirItensUmaLinha = nfceImprimirItensUmaLinha;
	}

    public String getNfceImprimirDescontoPorItem() {
		return this.nfceImprimirDescontoPorItem;
	}

	public void setNfceImprimirDescontoPorItem(String nfceImprimirDescontoPorItem) {
		this.nfceImprimirDescontoPorItem = nfceImprimirDescontoPorItem;
	}

    public String getNfceImprimirQrcodeLateral() {
		return this.nfceImprimirQrcodeLateral;
	}

	public void setNfceImprimirQrcodeLateral(String nfceImprimirQrcodeLateral) {
		this.nfceImprimirQrcodeLateral = nfceImprimirQrcodeLateral;
	}

    public String getNfceImprimirGtin() {
		return this.nfceImprimirGtin;
	}

	public void setNfceImprimirGtin(String nfceImprimirGtin) {
		this.nfceImprimirGtin = nfceImprimirGtin;
	}

    public String getNfceImprimirNomeFantasia() {
		return this.nfceImprimirNomeFantasia;
	}

	public void setNfceImprimirNomeFantasia(String nfceImprimirNomeFantasia) {
		this.nfceImprimirNomeFantasia = nfceImprimirNomeFantasia;
	}

    public String getNfceImpressaoTributos() {
		return this.nfceImpressaoTributos;
	}

	public void setNfceImpressaoTributos(String nfceImpressaoTributos) {
		this.nfceImpressaoTributos = nfceImpressaoTributos;
	}

    public BigDecimal getNfceMargemSuperior() {
		return this.nfceMargemSuperior;
	}

	public void setNfceMargemSuperior(BigDecimal nfceMargemSuperior) {
		this.nfceMargemSuperior = nfceMargemSuperior;
	}

    public BigDecimal getNfceMargemInferior() {
		return this.nfceMargemInferior;
	}

	public void setNfceMargemInferior(BigDecimal nfceMargemInferior) {
		this.nfceMargemInferior = nfceMargemInferior;
	}

    public BigDecimal getNfceMargemDireita() {
		return this.nfceMargemDireita;
	}

	public void setNfceMargemDireita(BigDecimal nfceMargemDireita) {
		this.nfceMargemDireita = nfceMargemDireita;
	}

    public BigDecimal getNfceMargemEsquerda() {
		return this.nfceMargemEsquerda;
	}

	public void setNfceMargemEsquerda(BigDecimal nfceMargemEsquerda) {
		this.nfceMargemEsquerda = nfceMargemEsquerda;
	}

    public Integer getNfceResolucaoImpressao() {
		return this.nfceResolucaoImpressao;
	}

	public void setNfceResolucaoImpressao(Integer nfceResolucaoImpressao) {
		this.nfceResolucaoImpressao = nfceResolucaoImpressao;
	}

    public Integer getNfceTamanhoFonteItem() {
		return this.nfceTamanhoFonteItem;
	}

	public void setNfceTamanhoFonteItem(Integer nfceTamanhoFonteItem) {
		this.nfceTamanhoFonteItem = nfceTamanhoFonteItem;
	}

    public String getRespTecCnpj() {
		return this.respTecCnpj;
	}

	public void setRespTecCnpj(String respTecCnpj) {
		this.respTecCnpj = respTecCnpj;
	}

    public String getRespTecContato() {
		return this.respTecContato;
	}

	public void setRespTecContato(String respTecContato) {
		this.respTecContato = respTecContato;
	}

    public String getRespTecEmail() {
		return this.respTecEmail;
	}

	public void setRespTecEmail(String respTecEmail) {
		this.respTecEmail = respTecEmail;
	}

    public String getRespTecFone() {
		return this.respTecFone;
	}

	public void setRespTecFone(String respTecFone) {
		this.respTecFone = respTecFone;
	}

    public String getRespTecIdCsrt() {
		return this.respTecIdCsrt;
	}

	public void setRespTecIdCsrt(String respTecIdCsrt) {
		this.respTecIdCsrt = respTecIdCsrt;
	}

    public String getRespTecHashCsrt() {
		return this.respTecHashCsrt;
	}

	public void setRespTecHashCsrt(String respTecHashCsrt) {
		this.respTecHashCsrt = respTecHashCsrt;
	}

    public Empresa getEmpresa() {
		return this.empresa;
	}

	public void setEmpresa(Empresa empresa) {
		this.empresa = empresa;
	}

		
}

File: retaguarda-software-house/delphi/retaguarda_sh/Model/NFe/NfeConfiguracao.pas
{*******************************************************************************
Title: T2Ti ERP 3.0                                                                
Description: Model relacionado à tabela [NFE_CONFIGURACAO] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2021 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************}
unit NfeConfiguracao;

interface

uses
  MVCFramework.Serializer.Commons, ModelBase;

type

  [MVCNameCase(ncLowerCase)]
  TNfeConfiguracao = class(TModelBase)
  private
    FId: Integer;
    FCertificadoDigitalSerie: string;
    FCertificadoDigitalCaminho: string;
    FCertificadoDigitalSenha: string;
    FTipoEmissao: Integer;
    FFormatoImpressaoDanfe: Integer;
    FProcessoEmissao: Integer;
    FVersaoProcessoEmissao: string;
    FCaminhoLogomarca: string;
    FSalvarXml: string;
    FCaminhoSalvarXml: string;
    FCaminhoSchemas: string;
    FCaminhoArquivoDanfe: string;
    FCaminhoSalvarPdf: string;
    FWebserviceUf: string;
    FWebserviceAmbiente: Integer;
    FWebserviceProxyHost: string;
    FWebserviceProxyPorta: Integer;
    FWebserviceProxyUsuario: string;
    FWebserviceProxySenha: string;
    FWebserviceVisualizar: string;
    FEmailServidorSmtp: string;
    FEmailPorta: Integer;
    FEmailUsuario: string;
    FEmailSenha: string;
    FEmailAssunto: string;
    FEmailAutenticaSsl: string;
    FEmailTexto: string;
    FNfceIdCsc: string;
    FNfceCsc: string;
    FNfceModeloImpressao: string;
    FNfceImprimirItensUmaLinha: string;
    FNfceImprimirDescontoPorItem: string;
    FNfceImprimirQrcodeLateral: string;
    FNfceImprimirGtin: string;
    FNfceImprimirNomeFantasia: string;
    FNfceImpressaoTributos: string;
    FNfceMargemSuperior: Extended;
    FNfceMargemInferior: Extended;
    FNfceMargemDireita: Extended;
    FNfceMargemEsquerda: Extended;
    FNfceResolucaoImpressao: Integer;
    FNfceTamanhoFonteItem: Integer;
    FRespTecCnpj: string;
    FRespTecContato: string;
    FRespTecEmail: string;
    FRespTecFone: string;
    FRespTecIdCsrt: string;
    FRespTecHashCsrt: string;
    FIdEmpresa: Integer;

  public
//    procedure ValidarInsercao; override;
//    procedure ValidarAlteracao; override;
//    procedure ValidarExclusao; override;

    [MVCColumnAttribute('ID', True)]
		[MVCNameAsAttribute('id')]
		property Id: Integer read FId write FId;
    [MVCColumnAttribute('CERTIFICADO_DIGITAL_SERIE')]
		[MVCNameAsAttribute('certificadoDigitalSerie')]
		property CertificadoDigitalSerie: string read FCertificadoDigitalSerie write FCertificadoDigitalSerie;
    [MVCColumnAttribute('CERTIFICADO_DIGITAL_CAMINHO')]
		[MVCNameAsAttribute('certificadoDigitalCaminho')]
		property CertificadoDigitalCaminho: string read FCertificadoDigitalCaminho write FCertificadoDigitalCaminho;
    [MVCColumnAttribute('CERTIFICADO_DIGITAL_SENHA')]
		[MVCNameAsAttribute('certificadoDigitalSenha')]
		property CertificadoDigitalSenha: string read FCertificadoDigitalSenha write FCertificadoDigitalSenha;
    [MVCColumnAttribute('TIPO_EMISSAO')]
		[MVCNameAsAttribute('tipoEmissao')]
		property TipoEmissao: Integer read FTipoEmissao write FTipoEmissao;
    [MVCColumnAttribute('FORMATO_IMPRESSAO_DANFE')]
		[MVCNameAsAttribute('formatoImpressaoDanfe')]
		property FormatoImpressaoDanfe: Integer read FFormatoImpressaoDanfe write FFormatoImpressaoDanfe;
    [MVCColumnAttribute('PROCESSO_EMISSAO')]
		[MVCNameAsAttribute('processoEmissao')]
		property ProcessoEmissao: Integer read FProcessoEmissao write FProcessoEmissao;
    [MVCColumnAttribute('VERSAO_PROCESSO_EMISSAO')]
		[MVCNameAsAttribute('versaoProcessoEmissao')]
		property VersaoProcessoEmissao: string read FVersaoProcessoEmissao write FVersaoProcessoEmissao;
    [MVCColumnAttribute('CAMINHO_LOGOMARCA')]
		[MVCNameAsAttribute('caminhoLogomarca')]
		property CaminhoLogomarca: string read FCaminhoLogomarca write FCaminhoLogomarca;
    [MVCColumnAttribute('SALVAR_XML')]
		[MVCNameAsAttribute('salvarXml')]
		property SalvarXml: string read FSalvarXml write FSalvarXml;
    [MVCColumnAttribute('CAMINHO_SALVAR_XML')]
		[MVCNameAsAttribute('caminhoSalvarXml')]
		property CaminhoSalvarXml: string read FCaminhoSalvarXml write FCaminhoSalvarXml;
    [MVCColumnAttribute('CAMINHO_SCHEMAS')]
		[MVCNameAsAttribute('caminhoSchemas')]
		property CaminhoSchemas: string read FCaminhoSchemas write FCaminhoSchemas;
    [MVCColumnAttribute('CAMINHO_ARQUIVO_DANFE')]
		[MVCNameAsAttribute('caminhoArquivoDanfe')]
		property CaminhoArquivoDanfe: string read FCaminhoArquivoDanfe write FCaminhoArquivoDanfe;
    [MVCColumnAttribute('CAMINHO_SALVAR_PDF')]
		[MVCNameAsAttribute('caminhoSalvarPdf')]
		property CaminhoSalvarPdf: string read FCaminhoSalvarPdf write FCaminhoSalvarPdf;
    [MVCColumnAttribute('WEBSERVICE_UF')]
		[MVCNameAsAttribute('webserviceUf')]
		property WebserviceUf: string read FWebserviceUf write FWebserviceUf;
    [MVCColumnAttribute('WEBSERVICE_AMBIENTE')]
		[MVCNameAsAttribute('webserviceAmbiente')]
		property WebserviceAmbiente: Integer read FWebserviceAmbiente write FWebserviceAmbiente;
    [MVCColumnAttribute('WEBSERVICE_PROXY_HOST')]
		[MVCNameAsAttribute('webserviceProxyHost')]
		property WebserviceProxyHost: string read FWebserviceProxyHost write FWebserviceProxyHost;
    [MVCColumnAttribute('WEBSERVICE_PROXY_PORTA')]
		[MVCNameAsAttribute('webserviceProxyPorta')]
		property WebserviceProxyPorta: Integer read FWebserviceProxyPorta write FWebserviceProxyPorta;
    [MVCColumnAttribute('WEBSERVICE_PROXY_USUARIO')]
		[MVCNameAsAttribute('webserviceProxyUsuario')]
		property WebserviceProxyUsuario: string read FWebserviceProxyUsuario write FWebserviceProxyUsuario;
    [MVCColumnAttribute('WEBSERVICE_PROXY_SENHA')]
		[MVCNameAsAttribute('webserviceProxySenha')]
		property WebserviceProxySenha: string read FWebserviceProxySenha write FWebserviceProxySenha;
    [MVCColumnAttribute('WEBSERVICE_VISUALIZAR')]
		[MVCNameAsAttribute('webserviceVisualizar')]
		property WebserviceVisualizar: string read FWebserviceVisualizar write FWebserviceVisualizar;
    [MVCColumnAttribute('EMAIL_SERVIDOR_SMTP')]
		[MVCNameAsAttribute('emailServidorSmtp')]
		property EmailServidorSmtp: string read FEmailServidorSmtp write FEmailServidorSmtp;
    [MVCColumnAttribute('EMAIL_PORTA')]
		[MVCNameAsAttribute('emailPorta')]
		property EmailPorta: Integer read FEmailPorta write FEmailPorta;
    [MVCColumnAttribute('EMAIL_USUARIO')]
		[MVCNameAsAttribute('emailUsuario')]
		property EmailUsuario: string read FEmailUsuario write FEmailUsuario;
    [MVCColumnAttribute('EMAIL_SENHA')]
		[MVCNameAsAttribute('emailSenha')]
		property EmailSenha: string read FEmailSenha write FEmailSenha;
    [MVCColumnAttribute('EMAIL_ASSUNTO')]
		[MVCNameAsAttribute('emailAssunto')]
		property EmailAssunto: string read FEmailAssunto write FEmailAssunto;
    [MVCColumnAttribute('EMAIL_AUTENTICA_SSL')]
		[MVCNameAsAttribute('emailAutenticaSsl')]
		property EmailAutenticaSsl: string read FEmailAutenticaSsl write FEmailAutenticaSsl;
    [MVCColumnAttribute('EMAIL_TEXTO')]
		[MVCNameAsAttribute('emailTexto')]
		property EmailTexto: string read FEmailTexto write FEmailTexto;
    [MVCColumnAttribute('NFCE_ID_CSC')]
		[MVCNameAsAttribute('nfceIdCsc')]
		property NfceIdCsc: string read FNfceIdCsc write FNfceIdCsc;
    [MVCColumnAttribute('NFCE_CSC')]
		[MVCNameAsAttribute('nfceCsc')]
		property NfceCsc: string read FNfceCsc write FNfceCsc;
    [MVCColumnAttribute('NFCE_MODELO_IMPRESSAO')]
		[MVCNameAsAttribute('nfceModeloImpressao')]
		property NfceModeloImpressao: string read FNfceModeloImpressao write FNfceModeloImpressao;
    [MVCColumnAttribute('NFCE_IMPRIMIR_ITENS_UMA_LINHA')]
		[MVCNameAsAttribute('nfceImprimirItensUmaLinha')]
		property NfceImprimirItensUmaLinha: string read FNfceImprimirItensUmaLinha write FNfceImprimirItensUmaLinha;
    [MVCColumnAttribute('NFCE_IMPRIMIR_DESCONTO_POR_ITEM')]
		[MVCNameAsAttribute('nfceImprimirDescontoPorItem')]
		property NfceImprimirDescontoPorItem: string read FNfceImprimirDescontoPorItem write FNfceImprimirDescontoPorItem;
    [MVCColumnAttribute('NFCE_IMPRIMIR_QRCODE_LATERAL')]
		[MVCNameAsAttribute('nfceImprimirQrcodeLateral')]
		property NfceImprimirQrcodeLateral: string read FNfceImprimirQrcodeLateral write FNfceImprimirQrcodeLateral;
    [MVCColumnAttribute('NFCE_IMPRIMIR_GTIN')]
		[MVCNameAsAttribute('nfceImprimirGtin')]
		property NfceImprimirGtin: string read FNfceImprimirGtin write FNfceImprimirGtin;
    [MVCColumnAttribute('NFCE_IMPRIMIR_NOME_FANTASIA')]
		[MVCNameAsAttribute('nfceImprimirNomeFantasia')]
		property NfceImprimirNomeFantasia: string read FNfceImprimirNomeFantasia write FNfceImprimirNomeFantasia;
    [MVCColumnAttribute('NFCE_IMPRESSAO_TRIBUTOS')]
		[MVCNameAsAttribute('nfceImpressaoTributos')]
		property NfceImpressaoTributos: string read FNfceImpressaoTributos write FNfceImpressaoTributos;
    [MVCColumnAttribute('NFCE_MARGEM_SUPERIOR')]
		[MVCNameAsAttribute('nfceMargemSuperior')]
		property NfceMargemSuperior: Extended read FNfceMargemSuperior write FNfceMargemSuperior;
    [MVCColumnAttribute('NFCE_MARGEM_INFERIOR')]
		[MVCNameAsAttribute('nfceMargemInferior')]
		property NfceMargemInferior: Extended read FNfceMargemInferior write FNfceMargemInferior;
    [MVCColumnAttribute('NFCE_MARGEM_DIREITA')]
		[MVCNameAsAttribute('nfceMargemDireita')]
		property NfceMargemDireita: Extended read FNfceMargemDireita write FNfceMargemDireita;
    [MVCColumnAttribute('NFCE_MARGEM_ESQUERDA')]
		[MVCNameAsAttribute('nfceMargemEsquerda')]
		property NfceMargemEsquerda: Extended read FNfceMargemEsquerda write FNfceMargemEsquerda;
    [MVCColumnAttribute('NFCE_RESOLUCAO_IMPRESSAO')]
		[MVCNameAsAttribute('nfceResolucaoImpressao')]
		property NfceResolucaoImpressao: Integer read FNfceResolucaoImpressao write FNfceResolucaoImpressao;
    [MVCColumnAttribute('NFCE_TAMANHO_FONTE_ITEM')]
		[MVCNameAsAttribute('nfceTamanhoFonteItem')]
		property NfceTamanhoFonteItem: Integer read FNfceTamanhoFonteItem write FNfceTamanhoFonteItem;
    [MVCColumnAttribute('RESP_TEC_CNPJ')]
		[MVCNameAsAttribute('respTecCnpj')]
		property RespTecCnpj: string read FRespTecCnpj write FRespTecCnpj;
    [MVCColumnAttribute('RESP_TEC_CONTATO')]
		[MVCNameAsAttribute('respTecContato')]
		property RespTecContato: string read FRespTecContato write FRespTecContato;
    [MVCColumnAttribute('RESP_TEC_EMAIL')]
		[MVCNameAsAttribute('respTecEmail')]
		property RespTecEmail: string read FRespTecEmail write FRespTecEmail;
    [MVCColumnAttribute('RESP_TEC_FONE')]
		[MVCNameAsAttribute('respTecFone')]
		property RespTecFone: string read FRespTecFone write FRespTecFone;
    [MVCColumnAttribute('RESP_TEC_ID_CSRT')]
		[MVCNameAsAttribute('respTecIdCsrt')]
		property RespTecIdCsrt: string read FRespTecIdCsrt write FRespTecIdCsrt;
    [MVCColumnAttribute('RESP_TEC_HASH_CSRT')]
		[MVCNameAsAttribute('respTecHashCsrt')]
		property RespTecHashCsrt: string read FRespTecHashCsrt write FRespTecHashCsrt;
    [MVCColumnAttribute('ID_EMPRESA')]
		[MVCNameAsAttribute('idEmpresa')]
		property IdEmpresa: Integer read FIdEmpresa write FIdEmpresa;
  end;

implementation

{ TNfeConfiguracao }



end.